packages/core/botpress/src/renderers/proactive.js

```diff
@@ -29,7 +29,7 @@ module.exports = ({ sendContent, db }) => {
    * @example
    * bp.renderers.sendToUser(event.user.id, '#!text-77734', { typingIndicators: false })
    */
-  const sendToUser = async (user, elementOrRenderer, data) => {
+  const sendToUser = async (user, elementOrRenderer, data, incoming) => {
```

Comment:
Can you explain what you do here? Because it's strong changes.

Reply:
The changes are not disruptive in any way, I am just passing extra optional params which wouldn't break any existing code

The reason is to be able to access the original incoming event (which carries pageId) in outgoing middleware. This is critical for certain logging operations, like submitting App Events to Facebook

These App Events are key to Facebook Analytics, Ads, and being able to run them profitably...

Reply:
We must to think about it. Because for me it's look bit confused

Reply:
How is this change creating any problems?